Wikipedia/UI-V5/WMFSaveButtonController.m

```diff
@@ -0,0 +1,83 @@
+//  Created by Monte Hurd on 8/19/15.2//  Copyright (c) 2015 Wikimedia Foundation. Provided under MIT-style license; please copy and modify!
+
+#import "WMFSaveButtonController.h"
+#import "MWKSavedPageList.h"
+#import "MWKArticle.h"
+#import "SessionSingleton.h"
+#import "MWKUserDataStore.h"
+
+@interface WMFSaveButtonController ()
+
+@property (strong, nonatomic) UIButton* button;
+
+@end
+
+@implementation WMFSaveButtonController
+
+- (instancetype)initWithButton:(UIButton*)button
+                         title:(MWKTitle*)title {
+    self = [super init];
+    if (self) {
+        self.button = button;
+        self.title  = title;
+        [self observeSavedPages];
+        [self updateSavedButtonState];
```

Comment:
Instead of manually calling `updateSavedButtonState` after every call to `observeSavedPages`, you can harden this "contract" by specifying the `NSKeyValueObservingOptionInitial` option, which invokes the KVO callback immediately after observation w/ the current value, thus causing you to update to the current state.

Either that, or add a call to `updateSavedButtonState` at the end of `observeSavedPages`.